src/core/mesh/qgsmeshlayer.h

```diff
@@ -203,14 +203,24 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer, public QgsAbstractProfileSo
     /**
      * Adds datasets to the mesh from file with \a path. Use the the time \a defaultReferenceTime as reference time is not provided in the file
      *
-     * \param path the path to the atasets file
+     * \param path the path to the datasets file
      * \param defaultReferenceTime reference time used if not provided in the file
      * \return whether the dataset is added
      *
      * \since QGIS 3.14
      */
     bool addDatasets( const QString &path, const QDateTime &defaultReferenceTime = QDateTime() );
 
+    /**
+     * Removes datasets from the mesh with given \a name.
+     *
+     * \param name of dataset group to remove
+     * \return whether the dataset is added
+     *
+     * \since QGIS 3.42
+     */
+    bool removeDatasets( const QString &name );
```

Comment:
Maybe call this `removeDataset` (singular) ?

Reply:
I followed the naming in original API here `addDatasets()` points to single file, adds single dataset group but it may actually be multiple datasets (time steps). 

In the same manner I used `removeDatasets()` that should be a complement to the above mentioned function. Single dataset group name is passed as parameter into this function but actually multiple datasets might be removed. 

Leaving this up to you, I am fine with either name @uclaros .